### PR TITLE
dispatch on abstract type H5DataStore

### PR DESCRIPTION
**Pull request by tbenst**
This allows for H5Sparse to use other notions of HDF5 datastores, like [HDF5.File, HDF5.Group, JldFile, JldGroup, Matlabv5File, and MatlabHDF5File](https://github.com/JuliaIO/HDF5.jl/blob/56c795cf876bcaeb37c9143fcccec01cbf0b1f30/src/HDF5.jl#L47).

All tests pass on my local machine.

### CODE REVIEW

**Comment by severinson** [comment]
Hey. Nice work. I'll happily merge the pull request. Should we add a type parameter to `H5SparseMatrixCSC` to avoid type instability though? I.e., the definition would be `struct H5SparseMatrixCSC{Tv, Ti<:Integer, Td<:HDF5.H5DataStore} <: SparseArrays.AbstractSparseMatrixCSC{Tv, Ti}`, and `fid::HDF5.H5DataStore` would become `fid::Td`.

**Comment by tbenst** [comment]
sure, let me know if I got that right!

File: src/H5Sparse.jl [hunk 1]
```diff
@@ -52,12 +52,13 @@ Matrix(A)           # Matrix
 ```
 
 """
-struct H5SparseMatrixCSC{Tv, Ti<:Integer} <: SparseArrays.AbstractSparseMatrixCSC{Tv, Ti}
-    fid::HDF5.H5DataStore          # Backing HDF5 file
+
+struct H5SparseMatrixCSC{Tv, Ti<:Integer, Td<:HDF5.H5DataStore} <: SparseArrays.AbstractSparseMatrixCSC{Tv, Ti}
+    fid::HDF5.H5DataStore   # Backing HDF5 file
```
Comment (severinson) [inline src/H5Sparse.jl]: Change this line to `fid::Td                 # Backing HDF5 store`

File: src/H5Sparse.jl [hunk 2]
```diff
@@ -52,12 +52,13 @@ Matrix(A)           # Matrix
...
     name::String            # Dataset name, i.e., data is stored in fid[name]
     rows::UnitRange{Int}    # Subset of rows stored in fid[name] accessible via this instance
     cols::UnitRange{Int}    # Subset of columns stored in fid[name] accessible via this instance
-    function H5SparseMatrixCSC(fid::HDF5.H5DataStore, name::AbstractString, rows::UnitRange{Int}, cols::UnitRange{Int}) where {Tv,Ti<:Integer}
+    function H5SparseMatrixCSC(fid::HDF5.H5DataStore, name::AbstractString, rows::UnitRange{Int}, cols::UnitRange{Int}) where {Tv,Ti<:Integer,Td<:HDF5.H5DataStore}
```
Comment (severinson) [inline src/H5Sparse.jl]: Remove `Td<:HDF5.H5DataStore`, so that it becomes `where {Tv,Ti<:Integer}`

File: src/H5Sparse.jl (hunk 1 above)
Comment (tbenst) [reply src/H5Sparse.jl]: done!

File: src/H5Sparse.jl (hunk 2 above)
Comment (tbenst) [reply src/H5Sparse.jl]: done!

**Comment by severinson** [comment]
Awesome. I'll merge it :)

**Pull request merged**